app/models/post.rb

```diff
@@ -260,10 +260,12 @@ def modify_author_reputation
     sc = saved_changes
     if sc.include?('deleted') && sc['deleted'][0] != sc['deleted'][1] && created_at >= 60.days.ago
       deleted = !!saved_changes['deleted']&.last
-      if deleted
-        user.update(reputation: user.reputation - Vote.total_rep_change(votes))
-      else
-        user.update(reputation: user.reputation + Vote.total_rep_change(votes))
+      if user
+        if deleted
+          user.update(reputation: user.reputation - Vote.total_rep_change(votes))
+        else
+          user.update(reputation: user.reputation + Vote.total_rep_change(votes))
+        end
       end
```

Comment:
Let's go for a cleaner option and use optional chaining (aka "safe navigation" in Ruby) instead to avoid the "hadouken" problem:

```ruby
if deleted
    user&.update(reputation: user.reputation - Vote.total_rep_change(votes))
else
    user&.update(reputation: user.reputation + Vote.total_rep_change(votes))
end 
```

Reply:
The `user` *should *always exist - deleted users aren't deleted, they just have a `deleted` flag set on them. Clearly that's failing in some cases, though. I'd add a SNO for the reputation as well. My suggestion would be:

```suggestion
      if deleted
        user&.update(reputation: (user&.reputation || 1) - Vote.total_rep_change(votes))
      else
        user&.update(reputation: (user&.reputation || 1) + Vote.total_rep_change(votes))
      end
```

Reply:
Oh, I wonder if it was a destroyed user.  I didn't try that test case yet.  (Would that make a difference?)

Reply:
Ok, that's god an error, though I got a 418, not a 500.

Repro: post something from a scratch account, log in as admin, *destroy* (not delete) the scratch account, then delete the post.